src/operator/tensor/ordering_op-inl.h

```diff
@@ -411,7 +414,8 @@ void TopKBackward_(const nnvm::NodeAttrs& attrs,
   } else if (kAddTo == req[0]) {
     // TODO(sxjscience) We can use AddTakeGrad in the future.
     // However, the current implementation of AddTakeGrad is not so efficient.
-    dummy_index = range<real_t>(0, sel_indices.shape_.Size());
+    mxnet_op::Kernel<range_fwd, xpu>::Launch(s, sel_indices.shape_.Size(), 1, 0.0f,
```

Comment:
repeat should be an integer.
also don't use dptr_. Use dptr<DType>()

Reply:
Thanks for your fix. But it is a `Tensor` not a `TBlob`, there is no such method `dptr<DType>()`. I think `dptr_` is fine here.

I chenge `1.0f` to `1` since `repeat` is an integer.

Reply:
Then you need to use a reinterprete_cast, otherwise there is no guarantee that the correct dtype will be used

Reply:
nvm, Tensor::dptr_ is actually typed